lib/modules/manager/flux/extract.ts

```diff
@@ -120,26 +124,29 @@ function resolveSystemManifest(
 function resolveResourceManifest(
   manifest: ResourceFluxManifest,
   helmRepositories: HelmRepository[],
+  helmCharts: HelmChart[],
   registryAliases: Record<string, string> | undefined,
 ): PackageDependency[] {
   const deps: PackageDependency[] = [];
   for (const resource of manifest.resources) {
     switch (resource.kind) {
       case 'HelmRelease': {
-        const dep: PackageDependency = {
-          depName: resource.spec.chart.spec.chart,
-          currentValue: resource.spec.chart.spec.version,
-          datasource: HelmDatasource.id,
-        };
+        if (resource.spec.chartRef?.kind === 'OCIRepository') {
+          // Can be skipped, version is handled via OCIRepository resource directly
+          break;
+        }
 
-        const matchingRepositories = helmRepositories.filter(
-          (rep) =>
-            rep.kind === resource.spec.chart.spec.sourceRef?.kind &&
-            rep.metadata.name === resource.spec.chart.spec.sourceRef.name &&
-            rep.metadata.namespace ===
-              (resource.spec.chart.spec.sourceRef.namespace ??
-                resource.metadata?.namespace),
+        const resolvedRelease = resolveHelmReleaseManifest(
+          resource,
+          helmRepositories,
+          helmCharts,
         );
+        if (!resolvedRelease) {
+          // invalid or incomplete HelmRelease spec, thus failed to extract the dependency itself
+          break;
```

Comment:
why break instead of continue?
```suggestion
          continue;
```